### PR TITLE
Update helpUrl for all tasks to point to new location of task reference

### PR DESCRIPTION
**Pull request by steved0x**
**Task name**: All tasks updated

**Description**: Updated the helpUrl in task.json and task.loc.json to point to the new location of the task reference, which has moved to https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/

When I published the new task reference last year, I also setup redirects so that customers clicking the old URLs would be redirected to the new location, so this PR isn't urgent, as the existing links still work by redirecting to the new location.

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N (although the automation I used to update these links checked each one to validate it was not a 404)

**Attached related issue:** (Y/N) N/A

**Checklist**:
- [N/A ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [N/A] Checked that applied changes work as expected

### CODE REVIEW

**Review by lucen-ms — APPROVED** [review]

**Comment by lucen-ms** [comment]
Approving for App Center

File: Tasks/MavenV2/task.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "name": "Maven",
     "friendlyName": "Maven",
     "description": "Build, test, and deploy with Apache Maven",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/build/maven",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/maven-v3",
```
Comment (jmyersmsft) [inline Tasks/MavenV2/task.json:6]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/maven-v2",
```

File: Tasks/MavenV4/task.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "name": "Maven",
     "friendlyName": "Maven",
     "description": "Build, test, and deploy with Apache Maven",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/build/maven",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/maven-v3",
```
Comment (jmyersmsft) [inline Tasks/MavenV4/task.json:6]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/maven-v4",
```

File: Tasks/NpmAuthenticateV0/task.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "name": "npmAuthenticate",
     "friendlyName": "npm authenticate (for task runners)",
     "description": "Don't use this task if you're also using the npm task. Provides npm credentials to an .npmrc file in your repository for the scope of the build. This enables npm task runners like gulp and Grunt to authenticate with private registries.",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/npm-authenticate",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/npm-v1-authenticate",
```
Comment (jmyersmsft) [inline Tasks/NpmAuthenticateV0/task.json:6]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/npm-authenticate-v0",
```

File: Tasks/NpmAuthenticateV0/task.loc.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "name": "npmAuthenticate",
   "friendlyName": "ms-resource:loc.friendlyName",
   "description": "ms-resource:loc.description",
-  "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/npm-authenticate",
+  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/npm-v0-authenticate",
```
Comment (jmyersmsft) [inline Tasks/NpmAuthenticateV0/task.loc.json:6]:
```suggestion
  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/npm-authenticate-v0",
```

File: Tasks/NuGetInstallerV0/task.json [hunk 5]
```diff
@@ -3,8 +3,8 @@
     "name": "NuGetInstaller",
     "friendlyName": "NuGet Installer",
     "description": "Installs or restores missing NuGet packages. Use NuGetAuthenticate@0 task for latest capabilities.",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget",
-    "helpMarkDown": "[More Information](https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget)",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2",
+    "helpMarkDown": "[More Information](https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2)",
```
Comment (jmyersmsft) [inline Tasks/NuGetInstallerV0/task.json:7]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-installer-v0",
    "helpMarkDown": "[More Information](https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-installer-v0)",
```

File: Tasks/NuGetInstallerV0/task.loc.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
   "name": "NuGetInstaller",
   "friendlyName": "ms-resource:loc.friendlyName",
   "description": "ms-resource:loc.description",
-  "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget",
+  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2",
```
Comment (jmyersmsft) [inline Tasks/NuGetInstallerV0/task.loc.json:6]:
```suggestion
  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-installer-v0",
```

File: Tasks/NuGetPackagerV0/task.json [hunk 7]
```diff
@@ -3,7 +3,7 @@
     "name": "NuGetPackager",
     "friendlyName": "NuGet packager",
     "description": "Deprecated: use the “NuGet” task instead. It works with the new Tool Installer framework so you can easily use new versions of NuGet without waiting for a task update, provides better support for authenticated feeds outside this organization/collection, and uses NuGet 4 by default.",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-packager-0",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-packager-0",
```
Comment (jmyersmsft) [inline Tasks/NuGetPackagerV0/task.json:6]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-packager-v0",
```

File: Tasks/NuGetPublisherV0/task.loc.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
   "name": "NuGetPublisher",
   "friendlyName": "ms-resource:loc.friendlyName",
   "description": "ms-resource:loc.description",
-  "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-publisher-0",
+  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-publisher-0",
```
Comment (jmyersmsft) [inline Tasks/NuGetPublisherV0/task.loc.json:6]:
```suggestion
  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-publisher-v0",
```

File: Tasks/NuGetRestoreV1/task.json [hunk 9]
```diff
@@ -3,8 +3,8 @@
     "name": "NuGetRestore",
     "friendlyName": "NuGet Restore",
     "description": "Restores NuGet packages in preparation for a Visual Studio Build step.",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget",
-    "helpMarkDown": "[Learn more about this task](https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget) or [see the NuGet documentation](https://docs.microsoft.com/nuget/)",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2",
+    "helpMarkDown": "[Learn more about this task](https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2) or [see the NuGet documentation](https://docs.microsoft.com/nuget/)",
```
Comment (jmyersmsft) [inline Tasks/NuGetRestoreV1/task.json:7]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-restore-v1",
    "helpMarkDown": "[Learn more about this task](https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-restore-v1) or [see the NuGet documentation](https://docs.microsoft.com/nuget/)",
```

File: Tasks/NuGetRestoreV1/task.loc.json [hunk 10]
```diff
@@ -3,7 +3,7 @@
   "name": "NuGetRestore",
   "friendlyName": "ms-resource:loc.friendlyName",
   "description": "ms-resource:loc.description",
-  "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget",
+  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2",
```
Comment (jmyersmsft) [inline Tasks/NuGetRestoreV1/task.loc.json:6]:
```suggestion
  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-restore-v1",
```

File: Tasks/NuGetToolInstallerV0/task.json [hunk 11]
```diff
@@ -3,7 +3,7 @@
     "name": "NuGetToolInstaller",
     "friendlyName": "NuGet tool installer",
     "description": "Acquires a specific version of NuGet from the internet or the tools cache and adds it to the PATH. Use this task to change the version of NuGet used in the NuGet tasks.",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-installer-0",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-installer-0",
```
Comment (jmyersmsft) [inline Tasks/NuGetToolInstallerV0/task.json:6]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-installer-v0",
```

File: Tasks/NuGetToolInstallerV0/task.loc.json [hunk 12]
```diff
@@ -3,7 +3,7 @@
   "name": "NuGetToolInstaller",
   "friendlyName": "ms-resource:loc.friendlyName",
   "description": "ms-resource:loc.description",
-  "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-installer-0",
+  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/package/prev-versions/nuget-installer-0",
```
Comment (jmyersmsft) [inline Tasks/NuGetToolInstallerV0/task.loc.json:6]:
```suggestion
  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-tool-installer-v0",
```

File: Tasks/NuGetV0/task.loc.json [hunk 13]
```diff
@@ -6,7 +6,7 @@
   "category": "Package",
   "author": "Microsoft Corporation",
   "preview": "true",
-  "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget",
+  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2",
```
Comment (jmyersmsft) [inline Tasks/NuGetV0/task.loc.json:9]:
```suggestion
  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-v0",
```

File: Tasks/NuGetV0/task.json [hunk 14]
```diff
@@ -6,7 +6,7 @@
     "category": "Package",
     "author": "Microsoft Corporation",
     "preview": "true",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-command-v2",
```
Comment (jmyersmsft) [inline Tasks/NuGetV0/task.json:9]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/nuget-v0",
```

File: Tasks/UniversalPackagesV0/task.loc.json [hunk 15]
```diff
@@ -4,7 +4,7 @@
   "friendlyName": "ms-resource:loc.friendlyName",
   "description": "ms-resource:loc.description",
   "author": "Microsoft Corporation",
-  "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks",
+  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks",
```
Comment (jmyersmsft) [inline Tasks/UniversalPackagesV0/task.loc.json:7]:
```suggestion
  "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/universal-packages-v0",
```

File: Tasks/UniversalPackagesV0/task.json [hunk 16]
```diff
@@ -4,7 +4,7 @@
     "friendlyName": "Universal packages",
     "description": "Download or publish Universal Packages",
     "author": "Microsoft Corporation",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks",
+    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks",
```
Comment (jmyersmsft) [inline Tasks/UniversalPackagesV0/task.json:7]:
```suggestion
    "helpUrl": "https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/universal-packages-v0",
```

**Review by jmyersmsft — CHANGES_REQUESTED** [review]

**Pull request closed without merging**